### PR TITLE
Adds a name to map schemas

### PR DESCRIPTION
**Pull request by rodesai**
This patch adds names to map schemas when serializing avro. Internally,
the avro converter will use this name to name the map entry type. This
fixes the issue where, when a schema has multiple maps with different
types, avro errors out because the same name is used for different
map entry record schemas

### CODE REVIEW

File: ksql-serde/src/test/java/io/confluent/ksql/serde/avro/KsqlGenericRowAvroSerializerTest.java [hunk 1]
```diff
@@ -329,6 +341,53 @@ public void shouldSerializeMap() {
...
+        = mapEntrySchema("KsqlDataSourceSchema_field0_inner0");
+    final org.apache.avro.Schema avroInnerSchema1 =
+        mapEntrySchema(
+            "KsqlDataSourceSchema_field0_inner1",
+            org.apache.avro.SchemaBuilder.unionOf().nullType().and().stringType().endUnion());
+    final org.apache.avro.Schema avroSchema =
+        org.apache.avro.SchemaBuilder.record("KsqlDataSourceSchema_field0")
+            .namespace("io.confluent.ksql.avro_schemas")
+            .fields()
+            .name("inner0").type().unionOf().nullType().and().array()
```
Comment (agavra) [inline ksql-serde/src/test/java/io/confluent/ksql/serde/avro/KsqlGenericRowAvroSerializerTest.java]: I'm a little confused - this makes me think that the schema is an array of maps, but then below when we create the KSQL schema it seems to be just a single map. Am I missing something?

File: ksql-serde/src/main/java/io/confluent/ksql/serde/avro/AvroDataTranslator.java [hunk 2]
```diff
@@ -128,7 +128,8 @@ private Schema buildAvroCompatibleSchema(final Schema schema,
             buildAvroCompatibleSchema(schema.keySchema(),
                 typeNameGenerator.with(TypeNameGenerator.MAP_KEY_NAME)),
             buildAvroCompatibleSchema(schema.valueSchema(),
-                typeNameGenerator.with(TypeNameGenerator.MAP_VALUE_NAME)));
+                typeNameGenerator.with(TypeNameGenerator.MAP_VALUE_NAME))
+        ).name(typeNameGenerator.name());
```
Comment (rodesai) [inline ksql-serde/src/main/java/io/confluent/ksql/serde/avro/AvroDataTranslator.java]: Actually this needs more thought - we could be breaking schema compatibility.

File: ksql-serde/src/test/java/io/confluent/ksql/serde/avro/KsqlGenericRowAvroSerializerTest.java (hunk 1 above)
Comment (rodesai) [reply ksql-serde/src/test/java/io/confluent/ksql/serde/avro/KsqlGenericRowAvroSerializerTest.java]: Avro only lets you represent maps with string keys natively. So to work around this, connect will represent maps as an array of key-value pairs.

Here's what the avro schema for this test looks like:
```
{
	"type": "record",
	"name": "KsqlDataSourceSchema_field0",
	"namespace": "io.confluent.ksql.avro_schemas",
	"fields": [{
		"name": "inner0",
		"type": ["null", {
			"type": "array",
			"items": {
				"type": "record",
				"name": "KsqlDataSourceSchema_field0_inner0",
				"fields": [{
					"name": "key",
					"type": ["null", "string"],
					"default": null
				}, {
					"name": "value",
					"type": ["null", "long"],
					"default": null
				}],
				"connect.named.map": "true"
			}
		}],
		"default": null
	}, {
		"name": "inner1",
		"type": ["null", {
			"type": "array",
			"items": {
				"type": "record",
				"name": "KsqlDataSourceSchema_field0_inner1",
				"fields": [{
					"name": "key",
					"type": ["null", "string"],
					"default": null
				}, {
					"name": "value",
					"type": ["null", "long"],
					"default": null
				}],
				"connect.named.map": "true"
			}
		}],
		"default": null
	}]
}
```

Comment (agavra) [reply ksql-serde/src/test/java/io/confluent/ksql/serde/avro/KsqlGenericRowAvroSerializerTest.java]: Thanks @rodesai - this helps a lot.

File: ksql-serde/src/main/java/io/confluent/ksql/serde/avro/AvroDataTranslator.java (hunk 2 above)
Comment (rodesai) [reply ksql-serde/src/main/java/io/confluent/ksql/serde/avro/AvroDataTranslator.java]: this is done.

**Comment by rodesai** [comment]
No, it's a client-side change only. And those artifacts (avro converter) come from snapshot versions, so once the upstream sr change gets merged, built, and the artifacts pushed, this PR's build should succeed.

**Pull request merged**